pkgs/development/libraries/jsoncpp/1.7.2/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv
+, fetchgit
+, cmake
+, python
+}:
+stdenv.mkDerivation rec {
+  name = "jsoncpp-${version}";
+  version = "1.7.2";
```

Comment:
Did you try just updating the provided jsoncpp? Usually the latest version goes in default.nix and any older versions have their own file.